src/fluxpy/utils/model.py

```diff
@@ -415,3 +419,309 @@ def sync_with_medium(model: cobra.Model, medium: Dict):
+        # [['A', 'B', 'C'], ['D', 'K']]
+        
+
+    def cofactors_abundance(self):
+        
+        self.merge_connected_combinations()
+                
+        # Count occurrences of each cofactor
+        flat_list = [item for sublist in self.cofactors_list_all_reactions for item in sublist]
+        element_counts = Counter(flat_list)  
+        self.sorted_counts = element_counts.most_common()
+        
+        print(self.sorted_counts)
+        
+        
+    def remove_reactions(self):
+        
+        for knockout_index in self.knockout_indices_all:
+            knockout_reaction = self.reactions_ids[knockout_index]
+            self.model.reactions.get_by_id(knockout_reaction).lower_bound = 0
+            self.model.reactions.get_by_id(knockout_reaction).upper_bound = 0
+                
+        
+    def filter_reactions(self):
+        
+        self.cofactors_abundance()
+        self.knockout_indices_all = []
+        
+        for group in self.groups:
+            keep_indices_group = []
+            cofactors_count_min = float('+inf')
+            
+            print(group)
+            
+            # find minimum number of cofactors across reactions of a single group
+            for reaction in group:
+                reaction_index = self.reactions_ids.index(reaction)   
+                reaction_cofactors = self.cofactors_list_all_reactions[reaction_index]
+                cofactors_count = len(reaction_cofactors)
+                
+                if cofactors_count < cofactors_count_min:
+                    cofactors_count_min = cofactors_count
+
+                print(reaction, reaction_cofactors, cofactors_count, cofactors_count_min)
+                
+            # find which reactions match the minimum number of cofactors
+            for reaction in group:
+                # find the index from the reaction list (general)
+                reaction_index = self.reactions_ids.index(reaction)                   
+                reaction_cofactors = self.cofactors_list_all_reactions[reaction_index]
+                cofactors_count = len(reaction_cofactors)
+   
+                # equal to minimum value
+                if cofactors_count == cofactors_count_min:
+                    keep_indices_group.append(reaction_index)
+                    
+                # greater than minimum value ==> knockout
+                else:
+                    self.knockout_indices_all.append(reaction_index)
+
+            
+            # if 2 or more reactions match the minimum number of cofactors
+            if len(keep_indices_group) > 1:
+                # find which reaction has the most abundant cofactor (general abundance from the model) 
+                # if the most abundant cofactor appears in all reactions, then check the next most abundant cofactor
+                
+                
+                def order_cofactors_by_abundance(sorted_counts_data, specific_reaction_cofactors):
```

Comment:
Functions like this and/or the `find_first_winning_sublist` that are not specific to the class, it is better to be out of it, so you can easily use them elsewhere; this way the class code is also easier to follow. 
Consider moving them to the `utils.py` script.